src/test/kotlin/org/elm/lang/core/resolve/ElmRecordFieldResolveTest.kt

```diff
@@ -265,4 +265,24 @@ main =
         --^                               
     ]
 """)
+
+    fun `test nested extension aliases with funcion in type variable passed through another variable via forward pipeline`() = checkByCode(
+            """
+infix left  0 (|>) = apR
+apR : a -> (a -> b) -> b
+apR x f = f x
+
+type alias R = { field : () }
+                  --X
+type alias Outer r = { r : Type (r -> r) }
+type Type a = Type a
+
+foo : Outer r -> Outer r
+foo r = r
+
+main : Outer R
+main =
+    { r = Type (\r -> { r | field = () }) } |> foo 
```

Comment:
It seems like there should be a smaller test case for this, but this is as much as I could minify it. For example, the problem isn't present if you use `<|` or a regular function call on this line.